test/e2e/persistent_volumes-disruptive.go

```diff
@@ -0,0 +1,263 @@
+		framework.SkipUnlessNodeCountIsAtLeast(MIN_NODES)
+		c = f.ClientSet
+		ns = f.Namespace.Name
+
+		// Start the NFS server pod.
+		if nfsServerPod == nil {
+			framework.Logf("[BeforeEach] Initializing NFS Server Pod")
+			nfsServerPod = startVolumeServer(c, nfsServerConfig)
+
+			framework.Logf("[BeforeEach] Configuring PersistentVolume")
+			nfsServerIP = nfsServerPod.Status.PodIP
+			nfsPVconfig = persistentVolumeConfig{
+				namePrefix: "nfs-",
+				pvSource: v1.PersistentVolumeSource{
+					NFS: &v1.NFSVolumeSource{
+						Server:   nfsServerIP,
+						Path:     "/exports",
+						ReadOnly: false,
+					},
+				},
+			}
+		}
+		// Get the first ready node IP that is not hosting the NFS pod.
+		if clientNodeIP == "" {
+			framework.Logf("Designating test node")
+			nodes := framework.GetReadySchedulableNodesOrDie(c)
+			for _, node := range nodes.Items {
+				if node.Name != nfsServerPod.Spec.NodeName {
+					clientNode = &node
+					break
+				}
+			}
+		}
+	})
+
+	AddCleanupAction(func() {
+		if nfsServerPod != nil && c != nil {
+			By("Deleting NFS server pod")
+			nfsServerPodCleanup(c, nfsServerConfig)
+			nfsServerPod = nil
+		}
+	})
+
+	Context("when kubelet restarts", func() {
+
+		var (
+			clientPod *v1.Pod
+			pv        *v1.PersistentVolume
+			pvc       *v1.PersistentVolumeClaim
+		)
+
+		BeforeEach(func() {
+			framework.Logf("Initializing test spec")
+			clientPod, pv, pvc = initTestCase(f, c, nfsPVconfig, ns, clientNode.Name)
+		})
+
+		AfterEach(func() {
+			framework.Logf("Tearing down test spec")
+			tearDownTestCase(c, f, ns, clientPod, pvc, pv)
+		})
+
+		// Test table housing the It() title string and test spec.  runTest is type testBody, defined at
+		// the start of this file.  To add tests, define a function mirroring the testBody signature and assign
+		// to runTest.
+		disruptiveTestTable := []disruptiveTest{
+			{
+				testItStmt: "Should test that a file written to the mount before kubelet restart is stat-able after restart.",
+				runTest:    testKubeletRestartsAndRestoresMount,
+			},
+			{
+				testItStmt: "Should test that a volume mount to a pod that is deleted while the kubelet is down unmounts when the kubelet returns.",
+				runTest:    testVolumeUnmountsFromDeletedPod,
+			},
+		}
+
+		// Test loop executes each disruptiveTest iteratively.
+		for _, test := range disruptiveTestTable {
+			func(t disruptiveTest) {
+				It(t.testItStmt+" [Disruptive]", func() {
+					By("Executing Spec")
+					t.runTest(c, f, clientPod, pvc, pv)
+				})
+			}(test)
+		}
+	})
+})
+
+// SPEC: Test that a volume mounted to a pod remains mounted after a kubelet restarts
+func testKubeletRestartsAndRestoresMount(c clientset.Interface, f *framework.Framework, clientPod *v1.Pod, pvc *v1.PersistentVolumeClaim, pv *v1.PersistentVolume) {
+	file := "/mnt/_SUCCESS"
+	_, err := podExec(clientPod, "touch "+file)
+	Expect(err).NotTo(HaveOccurred())
+
+	kubeletCommand(kRestart, c, clientPod)
```

Comment:
I tested this manually to make sure and found that the pod UID remains the same through a kubelet restart.  Left out re-getting the pod to avoid unnecessary api calls.